lib/collection/src/shards/mod.rs

```diff
@@ -52,3 +56,38 @@ pub async fn create_shard_dir(
         }
     }
 }
+
+/// Await for consensus to synchronize across all peers
+///
+/// This will take the current consensus state of this node. It then explicitly waits on all other
+/// nodes to reach the same (or later) consensus.
+///
+/// If awaiting on other nodes fails for any reason, this simply continues after the consensus
+/// timeout.
+///
+/// # Cancel safety
+///
+/// This function is cancel safe.
+async fn await_consensus_sync(
+    consensus: &dyn ShardTransferConsensus,
+    channel_service: &ChannelService,
+) {
+    let wait_until = tokio::time::Instant::now() + defaults::CONSENSUS_META_OP_WAIT;
+    let sync_consensus =
+        timeout_at(wait_until, consensus.await_consensus_sync(channel_service)).await;
+
+    match sync_consensus {
+        Ok(Ok(_)) => log::trace!("All peers reached consensus"),
+        // Failed to sync explicitly, waiting until timeout to assume synchronization
+        Ok(Err(err)) => {
+            log::warn!("All peers failed to synchronize consensus, waiting until timeout: {err}");
+            sleep_until(wait_until).await;
+        }
+        // Reached timeout, assume consensus is synchronized
+        Err(err) => {
+            log::warn!(
+                "All peers failed to synchronize consensus, continuing after timeout: {err}"
+            );
+        }
+    }
+}
```

Comment:
We had exactly this already, I just moved it here so we can use it in resharding.